files/build_templates/docker_image_ctl.j2

```diff
@@ -108,6 +108,37 @@ function preStartAction()
     fi
 {%- elif docker_container_name == "snmp" %}
     $SONIC_DB_CLI STATE_DB HSET 'DEVICE_METADATA|localhost' chassis_serial_number $(decode-syseeprom -s)
+
+    # Wait for system ready. Continue after timeout if we won't hear anything
+    # from DB. Read timeout from config file and add two extra minutes on top of
+    # it.
+    CONFFILE=system_health_monitoring_config.json
+    PLATFORM=$(sonic-cfggen -d -v "DEVICE_METADATA['localhost']['platform']")
+    TIMEOUT=$(cat /usr/share/sonic/device/$PLATFORM/$CONFFILE | jq ".timeout")
+    SYSTEM_READY_EVENT="SYSTEM_READY|SYSTEM_STATE"
+    CHECK_SR_EVENT="sonic-db-dump -n STATE_DB -k $SYSTEM_READY_EVENT"
+
+    # Set default if not found
+    if [[ -z $TIMEOUT ]]; then
+        MESSAGE="Failed to read timeout from config file, assuming default is \
+                10 minutes"
+        echo $MESSAGE
+        TIMEOUT=10
+    fi
+
+    # Add to extra minutes and convert to seconds and
```

Comment:
> What is the rationale behind adding extra two minutes?

@dgsudharsan The system ready status should be declared in `TIMEOUT` time, but I am adding two extra minutes just to poll it for a bit longer time to make sure it will overlap timeout time.

Reply:
Since you are polling every 15 seconds, shouldn't 1 minute be sufficient to handle overlap?